rfcs/convergence/file-naming.md

```diff
@@ -0,0 +1,148 @@
+
+**Why?**
+
+- Type names provide a consistent way to quickly identify what is in the file.
+- Type names make it easy to find a specific file type using an editor or IDE's fuzzy search techniques.
+- Type names provide pattern matching for any automated tasks.
+
+---
+
+- use `.tsx` only for files that contain view/rendering related logic
+
+**Why?**
+
+- will enforce to use JSX only where necessary (renderer only)
+- same as for Type Names
+
+---
+
+- (optional) use `.spec.` instead `.test.` suffix as we already follow behavioral driven approach (describes/it) rather than atomic `test()` approach
+  - as a next step, enable lint rule that will enforce one way or another within tests
+
+**Why?**
+
+- consistency/proper intent
+
+---
+
+- (optional) use kebab case for naming
+
+**Why?**
+
+- kebab case is more human readable especially in longer words
+- mitigates issues with git on various OS/case in/consistent files systems (eg. rename `helloWorld` to `helloworld`, commit, push, merge... now someone wants to change it back to `helloWorld`... )
+
+---
+
+<br/>
+
+**Proposed rules applied on current state:**
+
+| Current File Name   | Proposed File name  | Symbol Name                                                                         | Remarks                                                                                                                                                                                                                                                                                                                   |
+| ------------------- | ------------------- | ----------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| index.tsx           | index.ts            |                                                                                     | No changes from current                                                                                                                                                                                                                                                                                                   |
+| Example.tsx         | example.ts          | `export const Example = () => {..}`                                                 | To enforce rendering logic only within `.view.` type file/s. TS will enforce this by default. Not extremely picky on this one, we can use `.tsx` as well to match extension with test/story - priority is to be consistent                                                                                                |
```

Comment:
Personally, I find the naming `Example.tsx` to be intuitive since it tells me we're exporting an object/component/etc. In fact, this is how the naming convention works in Zig as well for instance - if a module is in fact a struct, then name it as capital such as `Example`; otherwise, lowercase.

Reply:
I hear you. Just note that this is primarily about having consistency.